src/bepasty/tests/test_rest_server.py

```diff
@@ -0,0 +1,816 @@
+#
+# REST api tests
+#
+
+import os
+import base64
+import threading
+import time
+import copy
+import hashlib
+import re
+from requests.auth import _basic_auth_str
```

Comment:
this does not look like a public function, can you search for a better way?

Reply:
I cant guess your favorite though, how about this?

```python
def add_auth(user, password, headers=None):
    req = Request('POST', 'http://dummy', auth=(user, password))

    headers = headers if headers is not None else {}
    headers['Authorization'] = req.prepare().headers['Authorization']
    return headers
```

Reply:
constructing a full Request ovjects seems a bit much, I thought there should be some public function somewhere that does what we need.

but functions starting with single or double underscore are usually not meant to be a public stable API, so that usually means "don't use this from outside".

Reply:
i looked around and found it rather embarrassing to not being able to find a reusable implementation of that in python stdlib, werkzeug, requests. everywhere it is just hardcoded inside some other stuff.

so, ignore my feedback and just use that internal function until we find something better.

Reply:
ok, i will make PR with _basic_auth_str later.

Reply:
thanks!